types/yaireo__tagify/dist/tagify.esm.d.ts

```diff
@@ -0,0 +1,12 @@
+// The package JSON "exports" field just redirects "./dist/tagify.esm.js" to "./index.d.ts"
+// This file exists solely for supporting Node 10, which does not support "exports" in package.json
+
+// Type tests are never run against a Node 10 environment.
+// Also, there's a lint rule that disallows relative imports in tests.
+// As a result, tests are never run against this file
+
+import Tagify = require("../index.js");
```

Comment:
I don't get it completely. So, this file not referenced in the package.json exports. Tests are not run against file. But it is still needed somehow?

Reply:
Try removing this file, then run `pnpm test yaireo__tagify`. You'll get the following:

![image](https://github.com/user-attachments/assets/571f51d0-e8d9-4049-99a9-f694e1a34adb)

As far as I understand it, this [repo uses attw](https://github.com/DefinitelyTyped/DefinitelyTyped?tab=readme-ov-file#arethetypeswrongcli-attw-checks) to check that types exist for all possible entry points. It also contains checks for `node10` resolution mode. This mode does not support the `exports` field, so it falls back to raw paths as they are in the package.

Reply:
Ok, I actually can't remember that attw was already part of the tests last time I updated the typings (or it just went through without any complains). So now, it's more than the basic type checking tests.

Would be nice, if attw would read the engine property of the original package (`"node": ">=16.15.0"`), than it could skip the node10 resolution mode. It's an already requested feature, but not yet capable of.

Reply:
> Would be nice, if attw would read the engine property of the original package (`"node": ">=16.15.0"`), than it could skip the node10 resolution mode. It's an already requested feature, but not yet capable of.

It was also new for me, I had to check their readme. But I think I remember reading somewere that just declaring `"node": ">=16.15.0"` isn't enough. Apparently, when you set `moduleResolution: node` in a `tsconfig.json`, that's interpreted as `node10`, and there's still quite a few packages that do that.

But either way it feels like the build setup is not perfect yet -- skipping node 10 would be great. For now, that seems what we need to do to pass to not fail the build.